integrationTests/multiShard/smartContract/executingSCCalls_test.go

```diff
@@ -0,0 +1,473 @@
+		advertiser.Close()
+		for _, n := range nodes {
+			n.node.Stop()
+		}
+	}()
+
+	proposerNodeShard1 := nodes[0]
+
+	// delay for bootstrapping and topic announcement
+	fmt.Println("Delaying for node bootstrap and topic announcement...")
+	time.Sleep(time.Second * 5)
+
+	senderAddressBytes := []byte("12345678901234567890123456789012")
+
+	// Minting sender account
+	createMintingForSenders(nodes, senderShard, [][]byte{senderAddressBytes}, senderMintingValue)
+
+	// should deploy smart contract -> we process a block containing only the sc deployment tx
+	scCode := "aaaa"
+	initialValueForInternalVariable := uint64(45)
+
+	contractTx := createTx(
+		t,
+		senderAddressBytes,
+		vm.CreateEmptyAddress().Bytes(),
+		senderNonce,
+		big.NewInt(0),
+		1,
+		100000,
+		scCode,
+		initialValueForInternalVariable,
+	)
+
+	err := proposerNodeShard1.txProcessor.ProcessTransaction(contractTx, generalRoundNumber)
+	assert.Nil(t, err)
+	_, err = proposerNodeShard1.accntState.Commit()
+	assert.Nil(t, err)
+
+	// Test that the gas for deploying the smart contract was substracted from the sender's account
+	acc, _ := proposerNodeShard1.node.GetAccount(hex.EncodeToString(senderAddressBytes))
+	expectedValue := big.NewInt(0)
+	expectedValue.Sub(senderMintingValue, big.NewInt(opGas*1))
+	assert.Equal(t, expectedValue, acc.Balance)
+	assert.Equal(t, senderNonce, acc.Nonce)
+
+	senderNonce++
+	generalRoundNumber++
+
+	// setting the sc deployment address (printed by the transaction processer)
+	scDeploymentAdddress, _ := hex.DecodeString("ca26d3e6152af91949295cc89f419413e08aa04ba2d5e1ed2b199b2ca8aabc2a")
+
+	// Now that the SC is deployed, we test a call from an account located in the first shard
+	addValue := uint64(100)
+	contractCallTx := createTx(
+		t,
+		senderAddressBytes,
+		scDeploymentAdddress,
+		senderNonce,
+		big.NewInt(0),
+		1,
+		100,
+		"Add",
+		addValue,
+	)
+
+	err = proposerNodeShard1.txProcessor.ProcessTransaction(contractCallTx, generalRoundNumber)
+	assert.Nil(t, err)
+	_, err = proposerNodeShard1.accntState.Commit()
+	assert.Nil(t, err)
+
+	// Test again that the gas for calling the smart contract was substracted from the sender's account
+	acc, _ = proposerNodeShard1.node.GetAccount(hex.EncodeToString(senderAddressBytes))
+	// Substract the gas price another time
+	expectedValue.Sub(expectedValue, big.NewInt(opGas*1))
+	assert.Equal(t, expectedValue, acc.Balance)
+	assert.Equal(t, senderNonce, acc.Nonce)
+
+
+	senderNonce++
+	generalRoundNumber++
+}
+
+// Test within a network of two shards the following situation
+// 1. Node in first shard deploys a smart contract -> we also make sure that the resulting smart contract address falls within the same shard
+// 2. Another account within the second shard calls the smart contract, we make sure the smart contract is updated and the gas
+//  is substracted from the caller's balance at the end
+func TestProcessSCCallsInMultiShardArchitecture_FirstShardReceivesCallFromSecondShard(t *testing.T) {
```

Comment:
function too long, could you split it in 2 ?

Reply:
I extracted the code that could be reused, if still not ok I will try refactor more of it